src/FilePreviewer.js

```diff
@@ -26,206 +51,208 @@ import ViewportContent from './ViewportContent';
-        setFile(setNewRotation(file));
-    };
-
-    /**
-     * Handler. Fit to screen.
-     *
-     * (This activates the "using fit to screen" mode, which
-     * makes the document or image resize based on the viewport size.)
-     *
-     * @return {void}
-     */
-    const onFitToScreen = () => {
-        setUsingFitToScreen(true);
-
-        // Get the "fit to screen" scale.
-        const newScale = originalSizes.map(originalSize =>
-            getFitToScreenScale(viewportSize, originalSize),
-        );
-
-        setFile(prevValue => R.assoc('scale', newScale, prevValue));
+  useEffect(() => {
+    // Reset all the attributes when the file prop changes.
+    setTotalPages(1);
+    setCurrentPage(0);
+    setOriginalSizes([]);
+    setFile(getFileProp(props));
+
+    // Unreference the viewport and container when the component is unmounting.
+    return () => {
+      viewportRef.current = null;
+      containerRef.current = null;
     };
+  }, [props.file]);
 
-    /**
-     * Handler. Download the current file, it can be a
-     * PDF or Image, as `url` or `base64` content.
-     *
-     * @return {void}
-     */
-    const onDownload = () => {
-        const url = file.url || `data:${file.mimeType};base64,${file.data}`;
-        return saveFile(url, file.name || 'download.pdf');
-    };
+  /**
+   * Handler. Scroll to the pervious page and update the index.
+   *
+   * @return {void}
+   */
+  const onPageUp = () => {
+    const previousIndex = R.clamp(0, totalPages, currentPage - 1);
 
-    /**
-     * Handler. Callback after the PDF gets
-     * processed successfully by `react-pdf`.
-     *
-     * @param  {Object} pdf
-     * @return {void}
-     */
-    const onLoadSuccess = pdf => {
-        // Wait until the original sizes gets calculated.
-        const promises = R.times(async i => {
-            // Wait until the original sizes gets calculated.
-            const page = await pdf.getPage(i + 1);
-            return R.pick(['width', 'height'], page.getViewport({scale: 1}));
-        }, pdf.numPages);
-
-        // Set the original sizes for this file.
-        Promise.all(promises).then(setOriginalSizes);
-
-        setTotalPages(pdf.numPages);
-        setUsingFitToScreen(true);
-    };
+    setCurrentPage(previousIndex);
 
-    // Set an effect to process the file resizing when the
-    // viewport sizes changes and it's using "fit to screen".
-    useEffect(() => {
-        // Check if it's using "fit to screen".
-        if (usingFitToScreen) {
-            // Get the "fit to screen" scale.
-            const newScale = originalSizes.map(originalSize =>
-                getFitToScreenScale(viewportSize, originalSize),
-            );
-
-            // Update the scale.
-            setFile(prevValue => R.assoc('scale', newScale, prevValue));
-        }
-    }, [usingFitToScreen, originalSizes, viewportSize]);
+    const previousPage = document.querySelector(
+      `div[data-pdfpage="${previousIndex}"]`,
```

Comment:
Where does the `data-pdfpage` attribute come from and will that ever change?

Reply:
Here: https://github.com/vendorpay/react-file-previewer/blob/master/src/PDFViewer.js#L15

Reply:
Can you add a comment in that file above that line with some context stating that if that is ever changed it needs to be changed in this file as well wherever it's referenced?